### PR TITLE
DOC: mention that `genlogistic` is one of several types of generalized logistic distributions

### PR DESCRIPTION
**Pull request by dschmitz89**
#### What does this implement/fix?
THere are several generalizations of the logistic distribution, `genlogistic` implements one of them. Acc. to [Wikipedia](https://en.wikipedia.org/wiki/Generalized_logistic_distribution#Type_I), it is often to referred as type 1/skew logistic.. This is now mentioned in the notes section.

### CODE REVIEW

**Comment by mdhaber** [comment]
Please post or email a screenshot of [1] so that I can verify. Also, please cite the source of "It is also referred to
    as the skew-logistic distribution."

**Comment by dschmitz89** [comment]
> Please post or email a screenshot of [1] so that I can verify. Also, please cite the source of "It is also referred to as the skew-logistic distribution", even if it's Wikipedia.

Screenshot from Johnson:

image

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -2766,12 +2766,22 @@ class genlogistic_gen(rv_continuous):
         f(x, c) = c \frac{\exp(-x)}
                          {(1 + \exp(-x))^{c+1}}
 
-    for real :math:`x` and :math:`c > 0`.
+    for real :math:`x` and :math:`c > 0`. In literature, different
+    generalizations of the logistic distribution can be found. This is the type 1
+    generalized logistic distribution according to [1]_. It is also referred to
+    as the skew-logistic distribution.
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:2772]:
```suggestion
    as the skew-logistic distribution [2]_.
```

**Review by mdhaber — APPROVED** [review]
Looks accurate. Hopefully docs build this time so we can see how they render.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -2766,12 +2766,22 @@ class genlogistic_gen(rv_continuous):
...
 
     `genlogistic` takes ``c`` as a shape parameter for :math:`c`.
 
     %(after_notes)s
 
+    References
+    ----------
+    .. [1] Johnson et al. "Continuous Univariate Distributions", Volume 2,
+           Wiley. 1995.
+    .. [2] "Generalized Logistic Distribution", Wikipedia
```
Comment (mdhaber) [inline scipy/stats/_continuous_distns.py:2782]:
```suggestion
    .. [2] "Generalized Logistic Distribution", Wikipedia,
```

**Comment by mdhaber** [comment]
Thanks @dschmitz89.

If you'd be willing to add a note that the `loggamma` distribution is actually the `expgamma` distribution, we could review and merge that quickly, too. I would also support adding the alias `expgamma` and deprecating `loggamma`.

I think `loglaplace` is also worth looking at. In quick tests, I was not able to confirm that either the logarithm or the exponential of a `loglaplace` random variable were distributed according to the `laplace` distribution.

**Comment by dschmitz89** [comment]
@mdhaber @j-bowhay : would it make sense to add the `expgamma` renaming to the depracation list #15765 ? I have never deprecated anything but could take a look.

**Comment by mdhaber** [comment]
It would need a note to the mailing list first, and I think that the PR with the deprecation would go in before adding to the tracker. The tracker reminds us to execute the removal of the deprecated behavior in two versions.

**Pull request merged**